src/test/ui/suggestions/suggest-move-types.stderr

```diff
@@ -13,70 +18,123 @@ LL | struct Al<'a, T, M: OneWithLifetime<A=(), T, 'a>> {
    |
 LL | struct B<T, U, V, M: Three<A=(), B=(), C=(), T, U, V>> {
-   |                            ----              ^  ^  ^ generic argument
-   |                            |                 |  |
-   |                            |                 |  generic argument
-   |                            |                 generic argument
-   |                            the first constraint is provided here
+   |                            ----  ----  ----  ^  ^  ^ generic argument
+   |                            |     |     |     |  |
+   |                            |     |     |     |  generic argument
+   |                            |     |     |     generic argument
+   |                            |     |     constraint
+   |                            |     constraint
+   |                            constraint
+   |
+help: move the constraints after the generic arguments
+   |
+LL | struct B<T, U, V, M: Three<T, U, V, A = (), B = (), C = ()>> {
+   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 error: generic arguments must come before the first constraint
   --> $DIR/suggest-move-types.rs:48:71
    |
 LL | struct Bl<'a, 'b, 'c, T, U, V, M: ThreeWithLifetime<A=(), B=(), C=(), T, U, V, 'a, 'b, 'c>> {
-   |                                                     ----              ^  ^  ^  ^^  ^^  ^^ generic argument
-   |                                                     |                 |  |  |  |   |
-   |                                                     |                 |  |  |  |   generic argument
-   |                                                     |                 |  |  |  generic argument
-   |                                                     |                 |  |  generic argument
-   |                                                     |                 |  generic argument
-   |                                                     |                 generic argument
-   |                                                     the first constraint is provided here
+   |                                                     ----  ----  ----  ^  ^  ^  ^^  ^^  ^^ generic argument
+   |                                                     |     |     |     |  |  |  |   |
+   |                                                     |     |     |     |  |  |  |   generic argument
+   |                                                     |     |     |     |  |  |  generic argument
+   |                                                     |     |     |     |  |  generic argument
+   |                                                     |     |     |     |  generic argument
+   |                                                     |     |     |     generic argument
+   |                                                     |     |     constraint
+   |                                                     |     constraint
+   |                                                     constraint
```

Comment:
We could change these to

```
error: generic arguments must come before the first constraint
  --> $DIR/suggest-move-types.rs:48:71
   |
LL | struct Bl<'a, 'b, 'c, T, U, V, M: ThreeWithLifetime<A=(), B=(), C=(), T, U, V, 'a, 'b, 'c>> {
   |                                                     ----  ----  ----  ^  ^  ^  ^^  ^^  ^^ generic arguments
   |                                                     |
   |                                                     constraints
```

What do you think?

Reply:
Yes, this would definitely be an improvement.

Reply:
I think this would only work because of the visual distinction between `----` and `^^^`. Otherwise, if you have `C A C A` then it doesn't work out, and I'm not sure how `.span_label(...)` will let you achieve that visually. I think if this doesn't add additional complexity then I'm in favor. Otherwise you're expending an awful amount on complexity on an error that arises rarely.